lib/job-iteration/interruption_adapters/good_job_adapter.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+begin
+  require "good_job"
```

Comment:
In addition to Ben's suggestion let's make sure folks are on the correct version of GoodJob for that new addition to work:
```suggestion
  require "good_job"
  gem "good_job", ">= 3.26"
```
If the requirement cannot be satisfied it'll raise `Gem::LoadError` explaining the need to upgrade:
```
> require "good_job"
=> false
> gem "good_job", ">= 4.0"
/Users/bdewater/.rbenv/versions/3.3.0/lib/ruby/3.3.0/bundler/rubygems_integration.rb:237:in 'block (2 levels) in replace_gem': can't activate good_job (>= 4.0), already activated good_job-3.26.1. Make sure all dependencies are added to Gemfile. (Gem::LoadError)
```